packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts

```diff
@@ -14,15 +15,17 @@ export const useDeployedContractInfo = <TContractName extends ContractName>(cont
     contractName as ContractName
   ] as Contract<TContractName>;
   const [status, setStatus] = useState<ContractCodeStatus>(ContractCodeStatus.LOADING);
-  const provider = useProvider({ chainId: scaffoldConfig.targetNetwork.id });
+  const publicClient = usePublicClient({ chainId: scaffoldConfig.targetNetwork.id });
 
   useEffect(() => {
     const checkContractDeployment = async () => {
       if (!deployedContract) {
         setStatus(ContractCodeStatus.NOT_FOUND);
         return;
       }
-      const code = await provider.getCode((deployedContract as Contract<TContractName>).address);
+      const code = await publicClient.getBytecode({
+        address: deployedContract.address as Address,
```

Comment:
Had to assert the type as `Address`  type imported from `viem` , It seems that due to `abitype` version mismatch in wagmi and viem we get the below error(even after configuring `AddressType`) : 
![Screenshot 2023-05-25 at 2 43 44 AM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/4089d6d4-e1a2-4d7b-bf68-ad74f43244f7)

Reply:
Isn't wagmi using viem? How come there's a mismatch in abitype?

Reply:
> Install wagmi and its viem peer dependency.

Actually [Wagmi](https://wagmi.sh/react/getting-started#installation) explicitly mentions installing `viem` as peer dependency :( which causes the mismatch because `viem`  uses diff version of `abitype` checkout -> https://github.com/wagmi-dev/wagmi/pull/2421 for more details

Reply:
Ah I understand, hope they can fix it by merging your PR 👍